src/lib/helpers/templateSource.ts

```diff
@@ -0,0 +1,32 @@
+export type Template = {
+  key?: string; // template slug, e.g. "magic-portfolio"
+  frameworks?: Array<{ 
+    key?: string;
+    providerRootDirectory?: string;
+  }>; // e.g. [{ key: "nextjs", providerRootDirectory: "./nextjs/magic-portfolio" }]
+  providerOwner?: string; // e.g. "appwrite"
+  providerRepositoryId?: string; // e.g. "templates-for-sites" or "template-for-store"
+  repository?: string; // if API already returns full URL, use it
+  repoUrl?: string;    // alias some builds may use
+  source?: string;     // alias some builds may use
+};
+
+export function getTemplateSourceUrl(template: Template): string | null {
+  // 1) Prefer a direct URL if the API already provides it
+  const direct = template.repository || template.repoUrl || template.source;
```

Comment:
From what I remember, there is only one way to construct git URL from resposne model we give, so some of those parameters are never existing.

@ItzNotABug can you please suggest ideal approach to build it here? This is response model: 
Functions: https://github.com/appwrite/appwrite/blob/7eaecd896e43cc1d34c710614bfee276e65a8570/src/Appwrite/Utopia/Response/Model/TemplateFunction.php#L89
Sites: https://github.com/appwrite/appwrite/blob/7eaecd896e43cc1d34c710614bfee276e65a8570/src/Appwrite/Utopia/Response/Model/TemplateSite.php#L69

Reply:
These vars don't exist on the response modal, so we can simply use the `protocol://host/owner/repository` string format. simple is fine.